client/my-sites/stats/stats-chart-tabs/utility.ts

```diff
@@ -0,0 +1,81 @@
+import { DateTime } from 'luxon';
+export { buildChartData } from './utility-legacy';
+
+type Period = 'day' | 'week' | 'month' | 'year';
+
+// Primarily used to format date within the chart tooltip.
+export function formatDate( dateString: string, period: Period ) {
+	if ( typeof dateString !== 'string' ) {
+		return null;
+	}
+
+	const dt = DateTime.fromFormat( dateString, 'yyyy-MM-dd' );
+	if ( ! dt.isValid ) {
+		return null;
+	}
+
+	switch ( period ) {
+		case 'day':
+			return dt.toLocaleString( DateTime.DATE_FULL );
+		case 'week': {
+			const startDate = dt.startOf( 'week' );
+			const endDate = startDate.endOf( 'week' );
+			return `${ startDate.toLocaleString( DateTime.DATE_SHORT ) } - ${ endDate.toLocaleString(
+				DateTime.DATE_SHORT
+			) }`;
+		}
+		case 'month':
+			return dt.toLocaleString( {
+				month: 'long',
+				year: 'numeric',
+			} );
+		case 'year':
+			return dt.toLocaleString( {
+				year: 'numeric',
+			} );
+		default:
+			return null;
+	}
+}
+
+// Returns the end date of a time interval including the desired date.
```

Comment:
It took me quite some time to understand what the function does. Could we please add a simple example here?

Reply:
Sorry for the confusion; this is an unfortunate remnant of the initial chart component logic 😞

I'll see what I can do to make this function easier to understand. This is the overall approach:

- Create a date window whose size is determined by `quantity * period`. 
    - For days/weeks/months, it's 30 days/weeks/months.
    - For years, the date window is 10 years.
    - This date window corresponds to the number of bars on the bar chart on the traffic page.
- The starting and ending dates for the current date window are determined by today's date.
    - For `day`, the window is [`2022-11-16`, `2022-12-16`] corresponding to [-30 days, today].
    - For `week`, the window is [`2022-05-20`, `2022-12-18`] corresponding to [Monday at -30 weeks, Sunday this week].

- Given the desired date, find the last date for the date window that the date belongs to.